atomicapp/display.py

```diff
@@ -0,0 +1,248 @@
+            return
+
+        if code == LOG_LEVELS['debug']:
+            self.logger.info(msg)
+            msg = "[DEBUG] %6d %0.2f %s" % (os.getpid(), time.time(), msg)
+        elif code == LOG_LEVELS['warning']:
+            self.logger.warning(msg)
+            msg = "[WARNING] %s" % msg
+        elif code == LOG_LEVELS['error']:
+            self.logger.error(msg)
+            msg = "[ERROR] %s" % msg
+        else:
+            self.logger.info(msg)
+            msg = "[INFO] %s" % msg
+
+        self._sysflush()
+        print(self._colorize(self._make_unicode(msg), color))
+
+    def _cockpit(self, msg, code, only):
+        """
+        Due to cockpit logging requirements, we will ONLY output logging that is designed as
+        display.info("foo bar", "cockpit")
+        """
+        if only is not "cockpit":
+            return
+
+        if self.verbose_level is not LOG_LEVELS['debug'] and code is LOG_LEVELS['debug']:
+            return
+
+        if code == LOG_LEVELS['debug']:
+            msg = "atomicapp.status.debug.message=%s" % msg
+        elif code == LOG_LEVELS['warning']:
+            msg = "atomicapp.status.warning.message=%s" % msg
+        elif code == LOG_LEVELS['error']:
+            msg = "atomicapp.status.error.message=%s" % msg
+        else:
+            msg = "atomicapp.status.info.message=%s" % msg
+
+        self._sysflush()
+        print(self._make_unicode(msg))
+
+    def _syslog(self, msg, code):
+        """
+        Outputs to the distro's default syslog.
+        """
+        if self.verbose_level is not LOG_LEVELS['debug'] and code is LOG_LEVELS['debug']:
+            return
+
+        handler = logging.handlers.SysLogHandler(address=LOG_SYSLOG)
+        self.logger.handlers = []
+        self.logger.addHandler(handler)
+
+        if code == LOG_LEVELS['debug']:
+            self.logger.info(msg)
+        elif code == LOG_LEVELS['warning']:
+            self.logger.warning(msg)
+        elif code == LOG_LEVELS['error']:
+            self.logger.error(msg)
+        else:
+            self.logger.info(msg)
+
+    def _sysflush(self):
+        """
+        Before each output, we check to see that we correctly flush out stderr
+        or stdout
+        """
+        if self.verbose_level is LOG_LEVELS['error'] or self.verbose_level is LOG_LEVELS['warning']:
+            sys.stderr.flush()
+        else:
+            sys.stdout.flush()
+
+    def _colorize(self, text, color):
+        """
+        Colorize based upon the color codes indicated.
+        """
+        return "\033[" + self.codeCodes[color] + "m" + text + "\033[0m"
+
+    def _make_unicode(self, input):
+        """
+        Convert all input to utf-8 for multi language support
+        """
+        if type(input) != unicode:
+            input = input.decode('utf-8')
+            return input
+        else:
+            return input
+
+
+def set_logging(args):
```

Comment:
Let's modify this to take vars as input. Looks like the few that we use are: `verbose`, `quiet`, and `logging_output`.

That way we aren't coupled to changes in cli/main.py.